FinSearchUnified/Subscriber/Frontend.php

```diff
@@ -195,7 +196,13 @@ public function onFrontendPostDispatch(Enlight_Event_EventArgs $args)
             return;
         }
 
-        $groupKey = Shopware()->Session()->get('sUserGroup', 'EK');
+        /** @var ContextService $context */
+        $context = Shopware()->Container()->get('shopware_storefront.context_service');
+        $groupKey = $context->getShopContext()->getCurrentCustomerGroup()->getKey();
+        if (!$groupKey) {
```

Comment:
That should not be needed. Shopware already defined this as fallback. See:

https://github.com/shopware/shopware/blob/v5.2.0/engine/Shopware/Bundle/StoreFrontBundle/Service/Core/ContextService.php#L48